src/status_im/common/device_network.cljs

```diff
@@ -0,0 +1,59 @@
+(ns status-im.common.device-network
+  (:require
+    ["@react-native-community/netinfo" :default net-info]
+    [native-module.core :as native-module]
+    [status-im.feature-flags :as ff]
+    [taoensso.timbre :as log]
+    [utils.re-frame :as rf]))
+
+(rf/reg-fx
+ :effects.device-network/listen-to-network-info
+ (fn []
+   (when net-info
+     (.addEventListener ^js net-info
```

Comment:
FYI This event listener logic is also present in this open PR https://github.com/status-im/status-mobile/pull/20730/files#diff-d1c789ab23b32d0ccbb4d5cf3dc0d6ba6fc33cbcc65c33d26ca9bde71956ef48R5